bftengine/src/communication/TlsTCPCommunication.cpp

```diff
@@ -811,17 +784,29 @@ class AsyncTlsConnection : public enable_shared_from_this<AsyncTlsConnection> {
   }
 
   void send(const char *data, uint32_t length) {
-    LOG_TRACE("enter, node " << _selfId << ", dest: " << _destId);
+    LOG_TRACE(_logger, "enter, node " << _selfId << ", dest: " << _destId);
 
     lock_guard<recursive_mutex> lock(_connectionsGuard);
     auto offset = prepare_output_buffer(MessageType::Regular,
                                         length);
     memcpy(_outBuffer + offset, data, length);
     write_async(_outBuffer, offset + length);
 
-    LOG_DEBUG("send exit, from: " << ", to: " << _destId << ", offset: " <<
-                                  offset << ", length: " << length);
-    LOG_TRACE("exit, node " << _selfId << ", dest: " << _destId);
+    if (_statusCallback && _isReplica) {
+      PeerConnectivityStatus pcs;
+      pcs.peerId = _selfId;
+      pcs.statusType = StatusType::MessageSent;
+
+      // pcs.statusTime = we dont set it since it is set by the aggregator
+      // in the upcoming version timestamps should be reviewed
+      _statusCallback(pcs);
+    }
+
+    LOG_DEBUG(_logger, "send exit, from: " << _selfId
+      << ", to: " << _destId
+      << ", offset: " << offset
+      << ", length: " << length);
+    LOG_TRACE(_logger, "exit, node " << _selfId << ", dest: " << _destId);
```

Comment:
This trace log is only going to print if the debug log has also printed. It doesn't add any new information. It should probably be removed.